# **INDIAN STARTUP ECOSYSTEM**

## Project Description
We embark on a journey of discovery as we leverage our data analysis expertise to uncover the untapped potential within the Indian startup ecosystem. This project is designed to not only decode the numbers but to distill insights that will guide our team towards a successful foray into this dynamic market.

## Scope of Work

- Conduct a thorough exploration of datasets, dissecting funding patterns, sectoral nuances, and geographical hotspots in the Indian startup landscap
- Analyze funding received by startups in india from 2018 to 2021



## Hypothesis 
**Null Hypothesis (H₀)**: There is no significant difference in the average funding received by startups and Location.

**Alternative Hypothesis (H₁)**: There are significant differences in the average funding received by startups and Location.

**Null Hypothesis (H0)**: There is no significant relationship between funding and the sector  

**Alternative Hypothesis (H1)**: There is a significant relationship between funding and the sector

## Questions 
1. How does funding vary across different industry sectors in India?
2. How does funding vary with the loaction of the start-ups
3. What is the relationship between the amount of funding and the stage of the company?
4. How have funding trends evolved between 2018 and 2021?
5. What are the most attractive sectors for investors?
6. Does the location of the company influence its sector?




# **DATA EXPLORATION, DATA UNDERSTANDING and DATA ANALYSIS**

In [2]:
# Load libraries
# Database connnection
import pyodbc     
from dotenv import dotenv_values    #import the dotenv_values function from the dotenv package

# Data manipulation
import pandas as pd
import numpy as np

# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# filter warnings
import warnings 
warnings.filterwarnings('ignore')

## **1. Loading and Inspection of Data**

**Load data from the SQL server**

In [3]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

In [4]:
# Create a connection string

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"
    
connection = pyodbc.connect(connection_string)



In [5]:
# sql query to get 2020 data. 
query_2020="SELECT * FROM dbo.LP1_startup_funding2020"

# sql query to get 2021 data. 
query_2021="SELECT * FROM dbo.LP1_startup_funding2021"

In [6]:
    # load 2021 data
data_2021=pd.read_sql(query_2021,connection)

    # load 2020 data
data_2020=pd.read_sql(query_2020,connection)

**Load CSV Files**

In [7]:
# load 2019 data
data_2019=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2019.csv')

    # load 2018 data
data_2018=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2018.csv')


### **2. Data Exploration and Understanding**

**Preview Each dataset**

In [8]:
# preview the rows and columns for the 2018 dataset
data_2018.sample(5)

Company Name                                           Industry  \
479      Knocksense                     News, Publishing, Social Media   
53   ONGO Framework  B2B, Information Services, Information Technology   
51    Dainik Jagran            Media and Entertainment, News, Outdoors   
27          Mintifi                        Financial Services, Lending   
148             OYO                Hospitality, Hotel, Leisure, Travel   

    Round/Series        Amount                          Location  \
479         Seed             —     Lucknow, Uttar Pradesh, India   
53         Angel       1000000  Hyderabad, Andhra Pradesh, India   
51          Seed   ₹40,000,000      Kanpur, Uttar Pradesh, India   
27          Seed       2000000        Mumbai, Maharashtra, India   
148     Series E  $100,000,000           Gurgaon, Haryana, India   

                                         About Company  
479    Knocksense is a fast growing lifestyle website.  
53   ONGO Framework is a B2B IT and digital solutio...  
51   Dainik Jagran is a media and communications gr...  
27             Online lending startup based in Mumbai.  
148  OYO is an Indian hospitality startup that oper...

In [9]:
# get a sample of 2019 dataset
data_2019.sample(5)

Company/Brand  Founded HeadQuarter        Sector  \
79      Zolostays   2015.0         NaN  Accomodation   
27  Fireflies .ai      NaN         NaN            AI   
75       Sportido   2015.0       Noida        Sports   
69     Nivesh.com   2016.0       Noida  Mutual Funds   
16         Byju's   2011.0         NaN        Edtech   

                                         What it does  \
79  It offers affordable housing apartments to you...   
27  Developer of an artificial intelligence-powere...   
75  It is a sports discovery app that helps to fin...   
69  The platform simplifies the process of investment   
16                   Provides online learning classes   

                                     Founders  \
79  Akhil Sikri, Nikhil Sikri, Sneha Choudhry   
27                                Sam Udotong   
75       Neeraj Aggarwala, Akhil Ramachandran   
69            Anurag Garg, Sridhar Srinivasan   
16                            Byju Raveendran   

                                             Investor     Amount($)     Stage  
79                                   Trifecta Capital    $7,000,000       NaN  
27                                    Canaan Partners    $5,000,000       NaN  
75  Inflection Point Ventures, SucSEED Venture Par...   Undisclosed  Series B  
69                                   Windrose Capital      $600,000       NaN  
16  South Africa’s Naspers Ventures, the CPP Inves...  $540,000,000       NaN

In [10]:
# get a sample of 2020 dataset
data_2020.sample(5)

Company_Brand  Founded HeadQuarter          Sector  \
933     Pepperfry   2013.0      Mumbai        E-market   
4      Gramophone   2016.0      Indore        AgriTech   
424   LEAD School   2012.0      Mumbai         EdtTech   
748    Convosight   2019.0   Hyderabad  Data Analytics   
787        Porter   2014.0      Mumbai       Logistics   

                                          What_it_does  \
933                  Online home furniture marketplace   
4    Gramophone is an AgTech platform enabling acce...   
424  Offers a high-end digital classroom system foc...   
748  It helps business gain insights from its commu...   
787                     Logistics and fleet management   

                                              Founders  \
933                       Ambareesh Murty, Ashish Shah   
4    Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
424                         Sumeet Mehta, Smita Deorah   
748                       Tarun Dhamija, Kartik Bansal   
787                           Pranav Goel, Uttam Digga   

                                Investor      Amount         Stage column10  
933                  Pidilite Industries  40000000.0      Series F     None  
4    Siana Capital Management, Info Edge    340000.0          None     None  
424    Westbridge Capital, Elevar Equity  28000000.0      Series C     None  
748     IvyCap Ventures, Sequoia Capital   4000000.0  Pre series A     None  
787              Lightstone Global Funds  18370000.0      Series D     None

In [11]:
# get a sample of 2021 dataset
data_2021.sample(5)

Company_Brand  Founded HeadQuarter              Sector  \
591     Powerplay   2020.0   Bangalore        Construction   
248   FanAnywhere   2021.0   Bangalore  Financial Services   
277       Bizongo   2015.0      Mumbai       B2B Ecommerce   
275         Dista   2017.0        Pune   Computer Software   
950     GramCover   2016.0       Noida           Insurance   

                                          What_it_does  \
591  Powerplay helps simplify end to end Constructi...   
248                             Celebrity NFT platform   
277  izongo is India’s largest B2B platform for mad...   
275  Dista is an AI-enabled location intelligence p...   
950  GramCover is a tech-enabled insurance marketpl...   

                                        Founders  \
591                    Shubham Goyal, Lesh Dixit   
248                  Varun Chaudhary, Amit Kumar   
277      Sachin Agrawal, Aniket Deb, Ankit Tomar   
275  Nishant Kumar, Shishir Gokhale, Deepak Garg   
950                               Dhyanesh Bhatt   

                                              Investor        Amount     Stage  
591                    Sequoia’s Surge, Accel Partners      $5200000      None  
248        Oasis Capital, Scorpio VC, DeltaHub Capital  $Undisclosed      Seed  
277                                       Tiger Global    $100000000  Series D  
275                                Pentathlon Ventures      $1200000      Seed  
950  Siana Capital, Inflexor Ventures, Stride Ventures      $7000000  Series A

**Shape of the data**

In [12]:
# get the number of rows and columns for the datasets
print(f"The 2018 dataset has {data_2018.shape[0]} rows and {data_2018.shape[1]} Columns\n")
print(f"The 2019 dataset has {data_2019.shape[0]} rows and {data_2019.shape[1]} Columns\n")
print(f"The 2020 dataset has {data_2020.shape[0]} rows and {data_2020.shape[1]} Columns\n")
print(f"The 2021 dataset has {data_2021.shape[0]} rows and {data_2021.shape[1]} Columns\n\n")

The 2018 dataset has 526 rows and 6 Columns

The 2019 dataset has 89 rows and 9 Columns

The 2020 dataset has 1055 rows and 10 Columns

The 2021 dataset has 1209 rows and 9 Columns




**Info of the data**

In [13]:
# overview of 2018 dataset
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [14]:
# overview of 2019 dataset
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [15]:
# overview of 2020 dataset
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [16]:
# overview of 2021 dataset
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


**Displaying datasets columns**

In [17]:
# 2021 data columns
data_2021.columns

Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage'],
      dtype='object')

In [18]:
# 2020 data columns

data_2020.columns

Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage', 'column10'],
      dtype='object')

In [19]:
# 2019 data columns

data_2019.columns

Index(['Company/Brand', 'Founded', 'HeadQuarter', 'Sector', 'What it does',
       'Founders', 'Investor', 'Amount($)', 'Stage'],
      dtype='object')

In [20]:
# 2018 data columns

data_2018.columns

Index(['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location',
       'About Company'],
      dtype='object')

## **Observations:**
**Issues with the data**

1. There is a discrepancy in the naming conventions between the columns in the 2018 and 2019 datasets compared to the 2020 and 2021 datasets.

2. The 2018 dataset exhibits some missing columns, contributing to an incomplete representation of the data.

3. Conversely, the 2020 dataset contains an additional column that appears to be extraneous and does not serve a meaningful purpose in our analysis.

**Course of Action:**

1. **Missing Column Engineering for 2018:**
   - We will address the absence of certain columns in the 2018 dataset by employing data engineering techniques to create and populate the missing columns, ensuring a comprehensive and consistent dataset.

2. **Column Name Standardization:**
   - To establish uniformity and coherence across all datasets, we will embark on a column renaming process for the 2018 and 2019 datasets. This action aims to align the naming conventions with those observed in the 2020 and 2021 datasets, facilitating seamless data integration and analysis.

3. **Extraneous Column Removal in 2020:**
   - The redundant column identified in the 2020 dataset will be removed, streamlining the dataset and eliminating unnecessary elements that do not contribute to the overall analysis objectives.

These actions collectively enhance the integrity, consistency, and completeness of the dataset, paving the way for a more robust and coherent analytical process.
ical process.







# **3. Data Cleaning**

In [21]:
data_2018

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed       250000       Bangalore, Karnataka, India   
1           Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2       Series A  ₹65,000,000           Gurgaon, Haryana, India   
3          Angel      2000000       Noida, Uttar Pradesh, India   
4           Seed            —  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C    225000000       Bangalore, Karnataka, India   
522     Series A            —           Haryana, Haryana, India   
523         Seed         7500        Mumbai, Maharashtra, India   
524         Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525     Series C     35000000        Chennai, Tamil Nadu, India   

                                         About Company  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[526 rows x 6 columns]

**Handling missing columns and currency signs in the 2018 dataset**

- The 2018 dataset has missing; 'founded', 'founders', and 'investor' columns


In [22]:
# Engineer missing columns for the 2018 dataset
columns_to_add = ['founded', 'founders', 'investor']
for column in columns_to_add:
    if column not in data_2018.columns:
        data_2018[column] = np.NaN

# Replace '₹', commas, '—', and "''" in 'Amount' column
data_2018['Amount'] = data_2018['Amount'].str.replace(',', '').str.replace('—', '').str.replace("''",'').replace('', np.nan)

# Conditionally apply multiplication only where '₹' is present
mask = data_2018['Amount'].str.contains('₹', na=False)
data_2018.loc[mask, 'Amount'] = data_2018.loc[mask, 'Amount'].str.replace('₹', '').astype(float) * 0.0146

**Data collection Year**

- There is need to add a column that represents the year each dataset was collected. This will help with handling the datasets after merging the dataframes 


In [23]:
# add year when data was collected column to every dataset
data_2018['data_year'] = pd.to_datetime('2018', format='%Y').year
data_2019['data_year'] = pd.to_datetime('2019', format='%Y').year
data_2020['data_year'] = pd.to_datetime('2020', format='%Y').year
data_2021['data_year'] = pd.to_datetime('2021', format='%Y').year

**Merge the dataframes**

**Notes**
- The function below concatenates the dataframes then renames the columns to ensure uniformity across the merged dataframe


In [28]:
#Define fuction to concatenate and rename columns
def concat_dfs(df_1, df_2, df_3, df_4):
    # Concatenate dataframes along the row axis and reset index
    df = pd.concat([df_1, df_2, df_3, df_4], ignore_index=True)

    # Rename columns before concatenating
    df_1.rename(columns={'Company Name': 'company_brand', 'Industry': 'sector', 'Round/Series': 'stage',
                         'Amount': 'amount($)', 'Location': 'location', 'About Company': 'about_company'},
                inplace=True)
    df_2.rename(columns={'Company/Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What it does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount($)': 'amount($)', 'Stage': 'stage'},
                inplace=True)
    df_3.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What_it_does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount': 'amount($)', 'Stage': 'stage'},
                inplace=True)
    df_4.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What_it_does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount': 'amount($)', 'Stage': 'stage'},
                inplace=True)

    return df

df = concat_dfs(data_2018, data_2019, data_2020, data_2021)


In [29]:
#Drop the extreneous column 10
df.drop('column10', axis=1, inplace= True)

**Cleaning 'Amount' column**

In [26]:
try:
    # Remove dollar sign
    df['amount($)'] = df['amount($)'].replace('\$', '', regex=True)

    # Remove commas
    df['amount($)'] = df['amount($)'].str.replace(',', '')

    # Remove specified values
    df['amount($)'] = df['amount($)'].replace(["Upsparks", 'undisclosed', 'Undisclosed', "ah! Ventures", 
                                               "Pre-series A", "ITO Angel Network LetsVenture", 
                                               "JITO Angel Network LetsVenture", "Series C", 'Seed', ','], '')

    # Convert the 'amount($)' column to numeric
    df['amount($)'] = pd.to_numeric(df['amount($)'])

except Exception as e:
    print(f'Error: {e}')



Error: 'amount($)'


In [30]:
df['amount($)'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 2879 entries, 0 to 2878
Series name: amount($)
Non-Null Count  Dtype 
--------------  ----- 
2474 non-null   object
dtypes: object(1)
memory usage: 22.6+ KB


**Cleaning data_year column**

In [33]:
# Convert the data_year column to date
df['data_year']=pd.to_datetime(df['data_year'], format='%Y')
df['data_year']=df['data_year'].dt.to_period('y')
# df['founded']=pd.to_datetime(df['founded']).dt.year

In [34]:
# check for nulls and duplicated
print(f"There are {df['data_year'].isna().sum()} Null values in the 'data_year' column")

There are 0 Null values in the 'data_year' column


In [35]:
df['data_year'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 2879 entries, 0 to 2878
Series name: data_year
Non-Null Count  Dtype        
--------------  -----        
2879 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 22.6 KB


**Cleaning 'founded' column**

In [ ]:
print(f"There are {df['founded'].isna().sum()} Null values in the 'founded' column")

There are 2844 and 769 Null values in the 'founded' column


**Notes**  
- There are 769 null values in the 'founded' column.  

- Since dropping the nulls will lead to a significant loss of our data, Backward fill will be used to fill the null values

In [31]:
# Fill missing values
df['founded']=df['founded'].fillna(method='bfill')

# convert to datetime
df['founded']=pd.to_datetime(df['founded'], format='%Y')

df['founded']=df['founded'].dt.to_period('y')

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype        
---  ------         --------------  -----        
 0   company_brand  2879 non-null   object       
 1   sector         2861 non-null   object       
 2   stage          1941 non-null   object       
 3   amount($)      2474 non-null   object       
 4   location       2765 non-null   object       
 5   about_company  2879 non-null   object       
 6   founded        2879 non-null   period[Y-DEC]
 7   founders       2334 non-null   object       
 8   investor       2253 non-null   object       
 9   data_year      2879 non-null   period[Y-DEC]
dtypes: object(8), period[Y-DEC](2)
memory usage: 225.1+ KB
